### DISCUSSION TITLE
I want to see that the close icon has focus.

### DISCUSSION THREAD

**Discussion by hgfrey** [primefaces/community#1031] [PrimeFaces] [answered]
Thanks to Melloware's help, I can now set the focus to the message close icon when an error occurs. The error message now closes reliably when I press Enter or Space. What is still missing is that you can see that the message close icon has the focus. This works as long as I use the application with the keyboard. But if I trigger the error with the mouse, the message-close-icon gets the focus, but the dark border is missing. I couldn't figure out what is responsible for the dark border being drawn.

Do you have any help for me? Thank you.

**Comment by melloware** [comment, marked as answer]
ahh here is why it uses ` :focus` psuedo CSS class and that can only be activated by real keyboard focus not JS focus().

See: https://stackoverflow.com/questions/69283396/what-can-cause-focus-visible-style-to-not-be-applied-when-using-element-focus

![image](https://github.com/primefaces/community/assets/4399574/6674ed35-d7b3-4bf4-882f-27697496ab34)


So you could style it manually with CSS to make it look like it has the focus like the CSS `:focus` pseudo is doing but that feels kinda hacky.

**Reply by hgfrey** [reply]
the hack works :-)
